tests/unit/test_job.py

```diff
@@ -5344,7 +5344,7 @@ def test_to_arrow(self):
         client = _make_client(project=self.PROJECT, connection=connection)
         job = self._make_one(self.JOB_ID, self.QUERY, client)
 
-        tbl = job.to_arrow()
+        tbl = job.to_arrow(create_bqstorage_client=False)
```

Comment:
Before this PR, a BigQuery Storage client was automatically created (a default) behind the scenes and actually hit the backend, but only to trigger a 401 Unauthorized error and fall back to the `tabledata.list` API that uses the (mocked) BigQuery client.

These `create_bqstorage_client=False` additions ensure that tests behave as before, i.e. using the (mocked) BigQuery client, but without a futile attempt with a real BQ Storage client.

Reply:
Thanks for clarifying